websites/D/Duolingo/presence.ts

```diff
@@ -1,296 +1,289 @@
-			return set("Choosing something to learn");
-		}
-		case "practice":
-			return set("Practicing everything learned");
-		case "skill": {
-			if (path.length < 3) return;
-
-			const lesson = path[2]
-				.replace(/([a-z]+)([A-Z]|-\d)/g, "$1 $2")
-				.replace(/-(\d)/g, "$1");
-			if (path.length >= 4) {
-				switch (path[3]) {
-					case "tips":
-						return set(`Reading tips for ${lesson}`);
-					case "practice":
-						return set(`Practicing ${lesson}`);
-					case "test":
-						return set(`Testing out of ${lesson}`);
-					// No default
-				}
-			}
-			return set(`In a lesson: ${lesson}`);
-		}
-		case "shop":
-			return set("In the shop");
-		case "words":
-			return set("Viewing all learned words");
-		case "stories": {
-			if (path.length < 2) return set("Choosing a story to read");
-
-			const phrases = document.querySelectorAll(
-				`body > ${"div > ".repeat(11)} .phrase`
-			);
-			if (phrases.length !== 0) {
-				return set(
-					`Reading ${Array.from(phrases)
-						.map(p => p.textContent)
-						.join(" ")}`
-				);
-			}
-			return set("Reading a story");
+			case path.includes("test"):
+				presenceData.details = `Passed ${language.name} jump ahead test`;
+				break;
+
+			default:
+				presenceData.details = `Finished ${language.name} lesson`;
 		}
-		case "mistakes-review":
-			return set("Reviewing past mistakes");
-		case "checkpoint": {
-			if (path.length < 4) return;
-
-			const checkpoint = parseInt(path[2], 10) + 1;
-			if (path[3] === "practice")
-				return set(`Practicing Checkpoint ${checkpoint}`);
-			else if (path[3] === "bigtest")
-				return set(`Taking the Checkpoint ${checkpoint} test`);
-			return;
+	} else {
+		const progressBarElement = document
+			.querySelector('div[style*="--web-ui_internal_progress-bar-value"]')
+			?.getAttribute("style");
+		if (progressBarElement) {
+			presenceData.state = `${makeProgressBar(
+				Number(
+					/--web-ui_internal_progress-bar-value: ([^;]+)%/.exec(
+						progressBarElement
+					)?.[1]
+				),
+				100,
+				presenceData.details.length / 3
+			)}`;
 		}
-		// No default
 	}
-
-	return false;
+	if (!user.lessonTimeStamp) user.lessonTimeStamp = newTimeStamp();
 }
 
-function checkStartingPages(path: string[]): boolean {
-	function set(state: string) {
-		presenceData.details = "Getting started";
-		presenceData.state = state;
-		return true;
-	}
+async function updateData() {
+	const state = JSON.parse(window.localStorage.getItem("duo.state")).state
+		.redux;
+	if (!state) return;
 
-	switch (path[0]) {
-		case "register":
-			return set("Choosing a language");
-		case "welcome":
-			return set("Answering some questions");
-		case "placement":
-			return set("Taking the placement test");
-		// No default
-	}
+	user.currentCourseId = state.user.currentCourseId ?? null;
+	if (user.currentCourseId) setLang(/_(.*?)_/.exec(user.currentCourseId)?.[1]);
 
-	return false;
+	const [showTime] = await Promise.all([
+		presence.getSetting<boolean>("timestamps"),
+	]);
```

Comment:
Since there is only one setting, this can be simplified for now.
```suggestion
	const showTime = await presence.getSetting<boolean>("timestamps");
```